### PR TITLE
Export to Excel restrictions relaxed.  Fixes #2155

### LINKED ISSUE #2155: Allow xts (and other object types) to be transferred to Excel
Hello,

In the Variable Explorer window, currently only data frames can be opened in Excel. There are other R types, which are widely used, and are quite similar to a data frame. Time series objects (package xts) come to mind as an example. Could these be supported as well?

Thanks in advance

### PR DESCRIPTION
**Pull request by karthiknadig**
If a variable can be coerced to data.frame, and length is greater than 1, export to excel button is shown for that variable.
image

### CODE REVIEW

File: src/Package/Impl/DataInspect/VariableViewModel.cs [hunk 1]
```diff
@@ -61,8 +61,7 @@ public sealed class VariableViewModel : RSessionDataObject, IIndexedItem {
                 ShowDetailCommandTooltip = Resources.ShowDetailCommandTooltip;
             }
 
-            const ViewerCapabilities tableCaps = (ViewerCapabilities.Table | ViewerCapabilities.List);
-            CanShowOpenCsv = CanShowDetail && result.CanCoerceToDataFrame && _detailsViewer.Capabilities.HasFlag(tableCaps) && result.Length > 0;
+            CanShowOpenCsv = result.CanCoerceToDataFrame && result.Length > 1;
```
Comment (int19h) [inline src/Package/Impl/DataInspect/VariableViewModel.cs]: Can you also add a check for typename "S4" here? Those will usually have a length of 1, but they can still redefine `as.data.frame` - and if they do, we should show export for them.

Comment (karthiknadig) [inline src/Package/Impl/DataInspect/VariableViewModel.cs]: will do.

---
In reply to: [73923629](https://github.com/Microsoft/RTVS/pull/2157#discussion_r73923629) [](ancestors = 73923629)

**Comment by int19h** [comment]
:+1:

File: src/Package/Impl/DataInspect/VariableViewModel.cs [hunk 2]
```diff
@@ -61,7 +61,7 @@ public sealed class VariableViewModel : RSessionDataObject, IIndexedItem {
                 ShowDetailCommandTooltip = Resources.ShowDetailCommandTooltip;
             }
 
-            CanShowOpenCsv = result.CanCoerceToDataFrame && result.Length > 1;
+            CanShowOpenCsv = result.CanCoerceToDataFrame && (result.Length > 1 || (result.Length == 1 && result.TypeName == "S4"));
```
Comment (int19h) [inline src/Package/Impl/DataInspect/VariableViewModel.cs]: No need to do the `==1` check for S4 - they are length 1 by default, but can override to mean something else, and there's no reason to suppress export for those that do override length (but provide `as.data.frame`).

**Comment by int19h** [comment]
:+1:

**Pull request merged**